keywords/MobileRestClient.py

```diff
@@ -1365,6 +1448,73 @@ def verify_docs_present(self, url, db, expected_docs, auth=None, timeout=CLIENT_
 
             break
 
+    def get_changes(self, url, db, since, auth, feed="longpoll", timeout=60):
+        """
+        Issues a longpoll changes request with a provided since and authentication.
+        The timeout is in seconds.
+        Returns a python dictionary of the changes response in the format:
+
+        {u'last_seq': u'2', u'results': [{u'changes': [], u'id': u'_user/adam', u'seq': 2}]}
+        """
+
+        # Convert to ms for the sync_gateway REST api
+        timeout *= 1000
+
+        auth_type = get_auth_type(auth)
+        server_type = self.get_server_type(url)
+
+        if server_type == ServerType.listener:
+            resp = self._session.get("{}/{}/_changes?feed={}&since={}".format(url, db, feed, since))
+
+        elif server_type == ServerType.syncgateway:
+            body = {
+                "feed": feed,
+                "since": since,
+                "timeout": timeout
+            }
+
+            log_info("Using POST data: {}".format(body))
+
+            if auth_type == AuthType.session:
+                resp = self._session.post("{}/{}/_changes".format(url, db), data=json.dumps(body), cookies=dict(SyncGatewaySession=auth[1]))
+            elif auth_type == AuthType.http_basic:
+                resp = self._session.post("{}/{}/_changes".format(url, db), data=json.dumps(body), auth=auth)
+            else:
+                resp = self._session.post("{}/{}/_changes".format(url, db), data=json.dumps(body))
+
+        log_r(resp)
+        resp.raise_for_status()
+        resp_obj = resp.json()
+
+        log_info("Found {} changes".format(len(resp_obj["results"])))
+
+        return resp_obj
+
+    def verify_doc_id_in_changes(self, url, db, expected_doc_id, auth=None):
+        """ Verifies 'expected_doc_id' shows up in the changes feed starting with a since of 0
+        if the doc is not found before CLIENT_REQUEST_TIMEOUT, an exception is raised. If it is found,
+        return the last sequence that includes the doc
+        """
+
+        start = time.time()
+
+        last_seq = 0
+        while True:
+
+            if time.time() - start > CLIENT_REQUEST_TIMEOUT:
+                raise TimeoutException("Verify Docs In Changes: TIMEOUT")
+
+            resp_obj = self.get_changes(url=url, db=db, since=last_seq, auth=auth)
+            doc_ids_in_changes = [change["id"] for change in resp_obj["results"]]
+
+            if expected_doc_id in doc_ids_in_changes:
+                last_seq = resp_obj["last_seq"]
+                log_info("{} found at last_seq: {}".format(expected_doc_id, last_seq))
+                return last_seq
+            else:
+                log_info("'{}' not found retrying ...".format(expected_doc_id))
+                time.sleep(1)
```

Comment:
For longpoll (which I think is what you're doing here), I don't think a long sleep (1s) is required between changes requests - you should be able to resubmit immediately, since the request will block if no changes are found.